elasticsearch_dsl/search.py

```diff
@@ -693,7 +693,8 @@ def count(self):
 
         d = self.to_dict(count=True)
         # TODO: failed shards detection
-        return es.count(index=self._index, body=d, **self._params)["count"]
+        resp = es.count(index=self._index, query=d.get("query", None), **self._params)
```

Comment:
I did not keep `body` because the tests used `mocker.spy` and since body is dynamically added in elasticsearch-py the mock did not see it. But I could also use a different testing strategy if you prefer.